core/testing/context.go

```diff
@@ -3,22 +3,32 @@ package coretesting
 import (
 	"context"
 
+	"google.golang.org/protobuf/runtime/protoiface"
+
+	"cosmossdk.io/core/event"
 	"cosmossdk.io/core/store"
 )
 
 type dummyKey struct{}
 
 func Context() context.Context {
 	dummy := &dummyCtx{
-		stores: map[string]store.KVStore{},
+		stores:      map[string]store.KVStore{},
+		events:      map[string][]event.Event{},
+		protoEvents: map[string][]protoiface.MessageV1{},
 	}
 
 	ctx := context.WithValue(context.Background(), dummyKey{}, dummy)
 	return ctx
 }
 
 type dummyCtx struct {
+	// maps store by the actor.
 	stores map[string]store.KVStore
+	// maps event emitted by the actor.
+	events map[string][]event.Event
```

Comment:
Can we retain ordering a bit better. i.e. rather than a map how about an array of events + the module they came from

Reply:
@aaronc currently it is possible to only extract events from the module that emitted them. We could change that ofc if needed.